components/style/selector_matching.rs

```diff
@@ -380,15 +380,15 @@ impl Stylist {
                             return true
                         }
                     }
-                    mq_eval_changed(rules, before, after)
+                    mq_eval_changed(&rules, before, after)
```

Comment:
Probably because of deref coercions. `&foo` can become `&****foo` for an arbitrary number of dereferences.

My mistake.